pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -87,11 +82,14 @@ func (w *PodWatcher) computeChanges(podList []*Pod) ([]*Pod, error) {
 
 		// Detect new containers
 		newContainer := false
-		for _, container := range pod.Status.Containers {
-			if _, foundContainer := w.lastSeen[container.ID]; foundContainer == false {
-				newContainer = true
+		for _, container := range pod.Status.AllContainers() {
+			// init containers are never ready, we check for ID
+			if !container.IsPending() {
```

Comment:
here we omit pending containers on purpose ? I think it's worth saying why, from a user standpoint.

Reply:
By pending container we're meaning that we do not have an container ID see https://github.com/DataDog/datadog-agent/pull/3372/files/7ac1802a5db5832c921388403df914e9d3dda1f6#diff-a6d11e3302647241d7a6eaba71954fc4R100

We can't do much if the container is not running 🤷‍♂

Reply:
okay ..

Reply:
I reworded the comment to make it more explicit

Reply:
👍